### PR TITLE
Use Breeze style in Flatpak

### PR DESCRIPTION
**Pull request by ilya-fedin**
Before
Снимок экрана в 2020-07-03 05-20-53
After
Снимок экрана в 2020-07-03 05-21-08

### CODE REVIEW

**Comment by ilya-fedin** [comment]
> I'm fine to override the default theme to be Breeze but what if someone actually uses Fusion on his, say, LXQt-based desktop?

I also thought about that, but Qt doesn't have any API to get the current style name nor to check if it is was by the user or is default fallback. All I can do is check the values of the variables QT_QPA_PLATFORMTHEME and QT_STYLE_OVERRIDE, but, probably, that would be even more ugly (that's why I didn't do it first).

> And what if Quaternion is packed with Snap another day?

Currently, recommended way for snapping - [kde-neon extension](https://snapcraft.io/docs/qt5-kde-applications). Their wrapper already sets theme.

File: client/main.cpp [hunk 1]
```diff
@@ -93,6 +99,24 @@ int main( int argc, char* argv[] )
...
+
+    // Set style to Breeze if there are no additional styles,
+    // since Qt applications defaults to a very ugly Fusion
+    // in Flatpak outside KDE
+    if (inFlatpak() && appStyles == referenceStyleList)
+    {
+        // Set icon theme as well to have uniform design
+        QApplication::setStyle("Breeze");
+        QIcon::setThemeName("breeze");
+        QIcon::setFallbackThemeName("breeze");
```
[the remark this answers is not in the document]
Comment (ilya-fedin) [reply client/main.cpp:117]: fixed

**Comment by KitsuneRal** [comment]
> Qt doesn't have any API to get the current style name nor to check if it is was by the user or is default fallback. All I can do is check the values of the variables QT_QPA_PLATFORMTHEME and QT_STYLE_OVERRIDE, but, probably, that would be even more ugly (that's why I didn't do it first).

Right, relying on environment variables for hints is even more fragile. However, for now this looks like we impose another default over the Qt's default with no means to revert to the Qt's default - and if I got it right then sorry it's not an option :) If we can't give users the flexibility to bring the theme back to Fusion, we can merely provide an easy and unobtrusive way to switch to Breeze from inside Quaternion (via a setting, or a button somewhere in the [settings dialog (to be merged)](https://github.com/quotient-im/Quaternion/pull/605)). Or, given that Quaternion is for technical audience anyway, just document how to override the theme with generic means (`-style` and `QT_STYLE_OVERRIDE`).

> > And what if Quaternion is packed with Snap another day?
> 
> Currently, the recommended way for snapping is [kde-neon extension](https://snapcraft.io/docs/qt5-kde-applications). Their wrapper already sets a theme.

Good to know, thanks.

**Comment by ilya-fedin** [comment]
> with no means to revert to the Qt's default

There are a check if additional themes is installed (e.g. org.kde.KStyle.Adwaita). And if some theme is installed, then theme won't be set.

> we can merely provide an easy and unobtrusive way to switch to Breeze from inside Quaternion

Hmm, just like Kdenlive:
image

**Comment by KitsuneRal** [comment]
> > with no means to revert to the Qt's default
> 
> There are a check if additional themes is installed (e.g. org.kde.KStyle.Adwaita). And if some theme is installed, then theme won't be set.

What if there are no additional themes but the user just wants to use Fusion? This can be a case at Windows, e.g.

> > we can merely provide an easy and unobtrusive way to switch to Breeze from inside Quaternion
> 
> Hmm, just like Kdenlive:
> image

Yeah, looks good to me.

**Comment by ilya-fedin** [comment]
> This can be a case at Windows, e.g.

Window doesn't have Breeze anyway...

**Comment by KitsuneRal** [comment]
> > This can be a case at Windows, e.g.
> 
> Windows doesn't have Breeze anyway...

Ok, this is actually irrelevant if we talk about Flatpak. But Adwaita (any extra theme actually) may easily be missing exactly in cases you mentioned above - LXQt, WM, ion etc.

**Comment by ilya-fedin** [comment]
Created a setting, didn't relocate inFlatpak since it is in #682 and I think it is better to rebase this PR when #682 will be merged

File: client/main.cpp [hunk 2]
```diff
@@ -93,6 +99,23 @@ int main( int argc, char* argv[] )
         QApplication::setFont(font);
     }
 
+#if defined Q_OS_UNIX && !defined Q_OS_MAC
+    using Quotient::SettingsGroup;
+    const auto useBreezeStyle = SettingsGroup("UI").value("use_breeze_style", inFlatpak()).toBool();
```
[the remark this answers is not in the document]
Comment (ilya-fedin) [reply client/main.cpp]: Fixed

**Comment by ilya-fedin** [comment]
Rebased on the current master

**Pull request merged**